Jetty 10 - Improve test reliability

joakime opened the pull request:
Some more test reliability fixes

joakime committed:
NoRouteToHostException is not a timeout failure.

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Don't care about stop failures in this test

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

sbordet commented on jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/ConnectTimeoutTest.java:
```diff
@@ -50,6 +51,13 @@ public void testConnectTimeout() throws Exception
+                    // Unable to connect to destination is not a failure.
+                    latch.countDown();
+                    return;
+                }
```
@joakime I don't know what to say about this. 

We have already called `assumeConnectTimeout()` which verified that connecting to that address _indeed_ throws a `SocketTimeoutException`.

Few lines later we do it via `client.connect()` and we get a different exception?
We should really have `if (docker.screwsUpNetworking())`?

joakime replied:
Let me try a non-routable, local only, IP instead.

joakime replied:
The 10.* network is how docker itself communicates among containers.

joakime replied:
The original test is flaky, even on my local machine (no docker) this test will fail about 15% of the time.
For various reasons: "No Route to Host", "Host unreachable".
I'll use an IANA recommended host instead.

joakime committed:
Try a different host for connect timeout

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
More consistent ConnectTimeoutTest
 + using IANA example.com (as it will route to a
   real machine, and will never have this
   specific port open)

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
More consistent ConnectTimeoutTest
 + using IANA example.com (as it will route to a
   real machine, and will never have this
   specific port open)

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
More consistent ConnectTimeoutTest
 + using IANA example.com (as it will route to a
   real machine, and will never have this
   specific port open)

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
More consistent ConnectTimeoutTest
 + using IANA example.com (as it will route to a
   real machine, and will never have this
   specific port open)

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
More consistent ConnectTimeoutTest
 + using IANA example.com (as it will route to a
   real machine, and will never have this
   specific port open)

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>
